src/ReactiveDomain.Testing/Messaging/ConcurrentMessageQueue.cs

```diff
@@ -56,11 +56,10 @@ public ConcurrentMessageQueue(string name, IEnumerable<T> items)
         }
         public void AssertEmpty()
         {
-            if (!IsEmpty)
-            {
-                TryPeek(out var msg);
+            if (IsEmpty) return;
+            if( TryPeek(out var msg))
                 throw new Exception($"{_name} Queue is not Empty. Instead {msg.GetType()} is next.");
-            }
+            throw new Exception($"{_name} Queue is not Empty.");
```

Comment:
Can we avoid duplication here?

Reply:
There is not duplication, @joshkempner .
In your previous review you asked what happens if the 'try' fails.  I expect it to NOT fail because the queue is not empty.
However, since you do have that concern, I added the boolean and a dedicated, separate message.
This code does the job and will be very helpful for our build process.
No duplicate lines.